disnake/guild.py

```diff
@@ -4218,3 +4223,142 @@ async def timeout(
+
+        Requires the :attr:`~Permissions.manage_guild` permission.
+
+        .. versionadded:: 2.6
+
+        Raises
+        ------
+        Forbidden
+            You do not have proper permissions to retrieve auto moderation rules.
+        NotFound
+            An auto moderation rule with the provided ID does not exist in the guild.
+        HTTPException
+            Retrieving the rule failed.
+
+        Returns
+        -------
+        :class:`AutoModRule`
+            The auto moderation rule.
+        """
+        data = await self._state.http.get_auto_moderation_rule(self.id, rule_id)
+        return AutoModRule(data=data, guild=self)
+
+    async def fetch_automod_rules(self) -> List[AutoModRule]:
+        """|coro|
+
+        Retrieves the guild's auto moderation rules.
+        See also :func:`~Guild.fetch_automod_rule`.
+
+        Requires the :attr:`~Permissions.manage_guild` permission.
+
+        .. versionadded:: 2.6
+
+        Raises
+        ------
+        Forbidden
+            You do not have proper permissions to retrieve auto moderation rules.
+        NotFound
+            The guild does not have any auto moderation rules set up.
+        HTTPException
+            Retrieving the rules failed.
+
+        Returns
+        -------
+        List[:class:`AutoModRule`]
+            The guild's auto moderation rules.
+        """
+        data = await self._state.http.get_auto_moderation_rules(self.id)
+        return [AutoModRule(data=rule_data, guild=self) for rule_data in data]
+
+    async def create_automod_rule(
+        self,
+        name: str,
+        *,
+        event_type: AutoModEventType,
+        trigger_type: AutoModTriggerType,
+        actions: Sequence[AutoModAction],
+        trigger_metadata: AutoModTriggerMetadata = None,
+        enabled: bool = False,
+        exempt_roles: Optional[Sequence[Snowflake]] = None,
+        exempt_channels: Optional[Sequence[Snowflake]] = None,
+        reason: Optional[str] = None,
+    ) -> AutoModRule:
+        """|coro|
+
+        Creates a new :class:`AutoModRule` for the guild.
+
+        You must have :attr:`.Permissions.manage_guild` permission to do this.
+
+        The maximum number of rules for each trigger type is limited, see the
+        `api docs <https://discord.com/developers/docs/resources/auto-moderation#auto-moderation-limits-per-trigger-type>`__
+        for more details.
+
+        .. versionadded:: 2.6
+
+        Parameters
+        ----------
+        name: :class:`str`
+            The rule name.
+        event_type: :class:`AutoModEventType`
+            The type of events that this rule will be applied to.
+        trigger_type: :class:`AutoModTriggerType`
+            The type of trigger that determines whether this rule's actions should run for a specific event.
+            If set to :attr:`~AutoModTriggerType.keyword` or :attr:`~AutoModTriggerType.keyword_preset`,
+            ``trigger_metadata`` must be set accordingly.
```

Comment:
This should be documented that it cannot be changed after creation.

Reply:
Not sure if that's really needed, it's fairly self-explanatory as there's no `trigger_type` on `AutoModRule.edit`.

Reply:
https://github.com/DisnakeDev/disnake/pull/530/commits/2715d73f76b38742838c9910648e80456de40a72